cristin-project/src/main/java/no/unit/nva/cristin/projects/common/CristinQueryBuilder.java

```diff
@@ -0,0 +1,613 @@
+        return this;
+    }
+
+    /**
+     * Setter unit id.
+     */
+    public CristinQueryBuilder withUnit(String unit) {
+        cristinQuery.setValue(PROJECT_UNIT, unit);
+        return this;
+    }
+
+    /**
+     * Setter a person's username in Cristin together with the institution id separated by ':'.
+     */
+    public CristinQueryBuilder withUser(String user) {
+        cristinQuery.setValue(USER, user);
+        return this;
+    }
+
+    /**
+     * End Setters * Start private functions.
+     */
+
+    private void assignDefaultValues() {
+        requiredMissing().forEach(key -> {
+            switch (key) {
+                case PATH_ORGANISATION:
+                    if (cristinQuery.containsKey(IDENTITY)) {
+                        cristinQuery.setPath(key, cristinQuery.getValue(IDENTITY));
+                        cristinQuery.pathParameters.remove(IDENTITY);
+                    }
+                    break;
+                case PATH_PROJECT:
+                    cristinQuery.setPath(key, EMPTY_STRING);
+                    break;
+                case LANGUAGE:
+                    cristinQuery.setValue(key, DEFAULT_LANGUAGE_CODE);
+                    break;
+                case PAGE_CURRENT:
+                    cristinQuery.setValue(key, PARAMETER_PAGE_DEFAULT_VALUE);
+                    break;
+                case PAGE_ITEMS_PER_PAGE:
+                    cristinQuery.setValue(key, PARAMETER_PER_PAGE_DEFAULT_VALUE);
+                    break;
+                default:
+                    System.out.println("Unknown key " + key);
+                    break;
+            }
+        });
+    }
+
+    private void setPath(String key, String value) {
+        cristinQuery.isNvaQuery = true;
+        var nonNullValue = nonNull(value) ? value : EMPTY_STRING;
+
+        if (key.equals(IDENTITY.getNvaKey())) {
+            withPathIdentity(nonNullValue);
+        } else if (key.equals(PATH_ORGANISATION.getNvaKey())) {
+            withPathOrganization(nonNullValue);
+        } else if (key.equals(PATH_PROJECT.getNvaKey()) || key.equals(PATH_PROJECT.getKey())) {
+            withPathProject(nonNullValue);
+        } else {
+            System.out.printf("CQB::SETPATH::INVALID_KEY -> [%s]-[%s]\n\r", key, value);
+            invalidKeys.add(key);
+        }
+    }
+
+    private void setValue(String key, String value) {
+        var qpKey = keyFromString(key,value);
+        if (!key.equals(qpKey.getKey()) && !cristinQuery.isNvaQuery && qpKey != INVALID) {
+            cristinQuery.isNvaQuery = true;
+        }
+        switch (qpKey) {
+            case PATH_ORGANISATION:
+                withPathOrganization(value);
+                break;
+            case PATH_PROJECT:
+            case IDENTITY:
+                withPathIdentity(value);
+                break;
+            case PROJECT_ORGANIZATION:
+                withOrganization(value);
+                break;
+            case BIOBANK:
```

Comment:
Should all these be populated with logic?

Reply:
You are correct, moved some of them to default setValue